### PR TITLE
Update nodejs and other web components to the latest

### PR DESCRIPTION
**Pull request by nagineni**
Update Node.js, iotivity-node and iot-rest-api-server to the latest.

Node.js:  from v4.5.0 to the latest stable version v6.11.0.
Iotivity-node: from 1.2.0-2 to 1.2.0-5
iot-rest-api-server: github master.

### CODE REVIEW

**Comment by nagineni** [comment]
@poussa @mythi PTAL

**Comment by nagineni** [comment]
I made a local build with this PR and did some quick testing on MinnowBoard. Everything seems to be working.

File: recipes-devtools/nodejs/nodejs_6.11.0.bb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 DESCRIPTION = "nodeJS Evented I/O for V8 JavaScript"
 HOMEPAGE = "http://nodejs.org"
 LICENSE = "MIT & BSD-2-Clause & BSD-3-Clause & ISC & CC-BY-3.0 & AFL-2.1 & Apache-2.0 & OpenSSL & Zlib & Artistic-2.0 & (BSD-3-Clause | GPLv2)"
-LIC_FILES_CHKSUM = "file://LICENSE;md5=8e3c01094f0fcb889b13f0354e52f914"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=41a3a0ccf7f515cac3377389dd8faac8"
```
Comment (mythi) [inline recipes-devtools/nodejs/nodejs_6.11.0.bb:4]: Please check why the license checksum changed and mention it in the commit message

Comment (nagineni) [reply recipes-devtools/nodejs/nodejs_6.11.0.bb:4]: Done.

**Review by mythi — APPROVED** [review]
@nagineni LGTM

**Comment by nagineni** [comment]
@mythi Thanks!

I will merge this PR now.

**Pull request merged**